File: testdata/bench/c/BenchmarkTest00002.c
#include <stdio.h>
#include <sqlite3.h>

void handle_request(const char *user_input) {
    char query[512];
    sqlite3 *db;
    sqlite3_open(":memory:", &db);
    snprintf(query, sizeof(query), "SELECT name FROM accounts WHERE user = '%s'", user_input);
    sqlite3_exec(db, query, NULL, NULL, NULL);
    sqlite3_close(db);
}

File: testdata/bench/kotlin/BenchmarkTest00002.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via string concatenation in Ktor
suspend fun handler00002(call: ApplicationCall) {
    val name = call.request.queryParameters["name"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val query = "SELECT * FROM users WHERE name = '" + name + "'"
    val rs = conn.createStatement().executeQuery(query)
    call.respondText(rs.toString())
}
